### PR TITLE
Merge Container Insights on Windows to aws-cwa-dev with cherry-picked commits

### PR DESCRIPTION
**Pull request by KlwntSingh**
**Description:** <Describe what has changed.>
This PR adds CI on Windows feature in `awscontainerinsightsreceiver`. This PR merges dev branch `aws-ci-windows` into `aws-ci-dev`

Ex. Adding a feature - Explain what this achieves.-->
This PR adds CI on Windows feature in `awscontainerinsightsreceiver`.

**Testing:** <Describe what testing was performed and which tests were added.>
1. Unit tests has been added
2. e2e testing.

**Documentation:** <Describe the documentation added.>
This commit adds documentation - https://github.com/amazon-contributing/opentelemetry-collector-contrib/pull/166

### CODE REVIEW

**Review by mitali-salvi — DISMISSED** [review]

File: receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go [hunk 1]
```diff
@@ -141,6 +142,12 @@ func (e *ebsVolume) addEBSVolumeMapping(zone *string, attachement *ec2.VolumeAtt
 func (e *ebsVolume) findNvmeBlockNameIfPresent(devName string) string {
 	// for nvme(ssd), there is a symlink from devName to nvme block name, i.e. /dev/xvda -> /dev/nvme0n1
 	// https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/nvme-ebs-volumes.html
+
+	// Windows does not support file system eg: /rootfs to get Nvme Block name.
+	// todo: Implement logic to identify Nvme devices on Windows. Refer https://docs.aws.amazon.com/AWSEC2/latest/WindowsGuide/nvme-ebs-volumes.html#identify-nvme-ebs-device
+	if runtime.GOOS == "windows" {
```
Comment (nathalapooja) [inline receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go:148]: Can we create "windows"  constant in constants util file and use it everywhere?

**Review by nathalapooja — APPROVED** [review]
Added a nit comment

**Review by mitali-salvi — DISMISSED** [review]

File: receiver/awscontainerinsightreceiver/internal/stores/utils.go [hunk 2]
```diff
@@ -96,6 +96,14 @@ func stringInRuneset(name, subset string) bool {
 }
 
 func TagMetricSource(metric CIMetric) {
+	if metric.GetTag(ci.OperatingSystem) == "windows" {
```
Comment (nathalapooja) [inline receiver/awscontainerinsightreceiver/internal/stores/utils.go:99]: Change this to constant.

File: receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+		}
+
+		// add scaling group name
+		tags[ci.AutoScalingGroupNameKey] = k.hostInfo.GetAutoScalingGroupName()
+
+		// add tags for EKS
+		tags[ci.ClusterNameKey] = k.hostInfo.GetClusterName()
+
+		// add tags for OS
+		tags[ci.OperatingSystem] = "windows"
```
Comment (nathalapooja) [inline receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go:134]: Change this to constant.

File: receiver/awscontainerinsightreceiver/internal/stores/utils_test.go [hunk 4]
```diff
@@ -201,3 +201,43 @@ func TestUtils_TagMetricSource(t *testing.T) {
...
+		"[\"kubelet\",\"pod\",\"calculated\"]",
+		"[\"kubelet\",\"calculated\"]",
+		"[\"kubelet\",\"pod\",\"calculated\"]",
+		"[\"kubelet\",\"calculated\"]",
+		"[\"kubelet\"]",
+	}
+	for i, mtype := range types {
+		tags := map[string]string{
+			ci.MetricType:      mtype,
+			ci.OperatingSystem: "windows",
```
Comment (nathalapooja) [inline receiver/awscontainerinsightreceiver/internal/stores/utils_test.go:234]: Change this to constant.

**Review by nathalapooja — CHANGES_REQUESTED** [review]

File: receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go (hunk 1 above)
Comment (KlwntSingh) [reply receiver/awscontainerinsightreceiver/internal/host/ebsvolume.go]: Addressed

File: receiver/awscontainerinsightreceiver/internal/stores/utils.go (hunk 2 above)
Comment (KlwntSingh) [reply receiver/awscontainerinsightreceiver/internal/stores/utils.go]: Addressed

File: receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go (hunk 3 above)
Comment (KlwntSingh) [reply receiver/awscontainerinsightreceiver/internal/k8swindows/k8swindows.go]: Addressed

File: receiver/awscontainerinsightreceiver/internal/stores/utils_test.go (hunk 4 above)
Comment (KlwntSingh) [reply receiver/awscontainerinsightreceiver/internal/stores/utils_test.go]: Addressed

**Review by nathalapooja — APPROVED** [review]

**Review by mitali-salvi — APPROVED** [review]

**Pull request merged**